app/src/main/java/comp3350/mealbuddy/objects/Day.java

```diff
@@ -104,9 +104,89 @@ public String getMealString(MealTimeType MT) {
         return getMealTime(MT).toString();
     }
 
+    /*
+     * getExerciseString
+     * returns string interpretation of the exercises.
+     * Return:
+     *     returns the string interpretation.
+     */
+    public String getExerciseString() {
+        String exerciseString= "";
+        for (Exercise exercise : exercises) {
+            exerciseString += exercise.toString() + "\n";
+        }
+        return exerciseString;
+    }
+
+    /*
+     * addToMeal
+     * adds a food to the meal.
+     * Parameters:
+     *     @param MT - a mealtime enum value. {BREAKFAST, LUNCH, DINNER, SNACK}
+     *     @param edible - an edible to add to the meal
+     *     @param quantity - the quantity of the edible to add to the meal
+     * Return:
+     *     returns the string interpretation.
+     */
     public void addToMeal(MealTimeType MT, Edible edible, int quantity){
         getMealTime(MT).add(edible, quantity);
     }
 
+    /*
+     * addExercise
+     * add an exercise to the day
+     * Parameters:
+     *     @param exercise - the exercise to add
+     */
+    public void addExercise(Exercise exercise) {
+        if (exercises.contains(exercise)) {
+            double oldDuration = exercises.get(exercises.indexOf(exercise)).duration;
+            exercises.remove(exercise);
+            exercises.add(new Exercise(exercise.name, exercise.duration + oldDuration, exercise.intensity ));
+        } else {
+            exercises.add(exercise);
+        }
+    }
+
+    /*
+     * removeExercise
+     * remove an exercise from the day
+     * Parameters:
+     *     @param exercise - the exercise to remove
+     */
+    public void removeExercise(Exercise exercise) {
+        exercises.remove(exercise);
+    }
+
+    /*
+     * addGoal
+     * add a goal to the day
+     * Parameters:
+     *     @param goal - the goal to add
+     */
+    public void addGoal(Goal goal) {
+        if (!goals.contains(goal)) {
+            goals.add(goal);
+        }
+    }
+
+    /*
+     * removeGoal
+     * remove a goal from the day
+     * Parameters:
+     *     @param goal - the goal to remove
+     */
+    public void removeGoal(Goal goal) {
+        goals.remove(goal);
+    }
+
+    public ArrayList<Exercise> getExercises() {
+        return exercises;
+    }
+
+    public ArrayList<Goal> getGoals() {
+        return goals;
+    }
```

Comment:
Realistically this defeats the purpose of having the lists private in the first place since i can do a getExercise list and then just add a duplicate exercise there.  Really what we should do is be returning an iterator of the list.  return exercises.iterator() with return type Iterator<exercise>.  I'm fine with leaving it like this though but ideally the iterator is better.